tests/utils/test_cache_utils.py

```diff
@@ -137,39 +158,128 @@ def test_static_cache_mha_mqa_gqa(self):
+                (
+                    1,
+                    config.num_key_value_heads,
+                    1,
+                    config.hidden_size // config.num_attention_heads,
+                ),
                 device=torch_device,
             )
             return random_keys, random_values
 
         mha_config = LlamaConfig(num_attention_heads=32)
-        mha_static_cache = StaticCache(config=mha_config, batch_size=1, max_cache_len=10, device=torch_device)
+        mha_static_cache = StaticCache(
+            config=mha_config, batch_size=1, max_cache_len=10, device=torch_device
+        )
         cached_keys, cached_values = mha_static_cache.update(
-            *_random_kvs(mha_config), 0, cache_kwargs={"cache_position": torch.arange(1).to(torch_device)}
+            *_random_kvs(mha_config),
+            0,
+            cache_kwargs={"cache_position": torch.arange(1).to(torch_device)},
         )
         self.assertTrue(cached_keys.shape == (1, 32, 10, 128))
         self.assertTrue(cached_values.shape == (1, 32, 10, 128))
 
         gqa_config = LlamaConfig(num_attention_heads=32, num_key_value_heads=4)
-        gqa_static_cache = StaticCache(config=gqa_config, batch_size=1, max_cache_len=10, device=torch_device)
+        gqa_static_cache = StaticCache(
+            config=gqa_config, batch_size=1, max_cache_len=10, device=torch_device
+        )
         cached_keys, cached_values = gqa_static_cache.update(
-            *_random_kvs(gqa_config), 0, cache_kwargs={"cache_position": torch.arange(1).to(torch_device)}
+            *_random_kvs(gqa_config),
+            0,
+            cache_kwargs={"cache_position": torch.arange(1).to(torch_device)},
         )
         self.assertTrue(cached_keys.shape == (1, 4, 10, 128))
         self.assertTrue(cached_values.shape == (1, 4, 10, 128))
 
         mqa_config = LlamaConfig(num_attention_heads=32, num_key_value_heads=1)
-        mqa_static_cache = StaticCache(config=mqa_config, batch_size=1, max_cache_len=10, device=torch_device)
+        mqa_static_cache = StaticCache(
+            config=mqa_config, batch_size=1, max_cache_len=10, device=torch_device
+        )
         cached_keys, cached_values = mqa_static_cache.update(
-            *_random_kvs(mqa_config), 0, cache_kwargs={"cache_position": torch.arange(1).to(torch_device)}
+            *_random_kvs(mqa_config),
+            0,
+            cache_kwargs={"cache_position": torch.arange(1).to(torch_device)},
         )
         self.assertTrue(cached_keys.shape == (1, 1, 10, 128))
         self.assertTrue(cached_values.shape == (1, 1, 10, 128))
 
+    @slow
+    @require_read_token
+    def test_dynamic_cache_exportability(self):
+        if not is_torch_greater_or_equal_than_2_6:
+            self.skipTest(reason="This test requires torch >= 2.6 to run.")
+
+        model = AutoModelForCausalLM.from_pretrained("Qwen/Qwen2-7B")
```

Comment:
Since this test doesn't care about the exact outputs of the model, we should use a tiny test model instead :) E.g. `hf-internal-testing/tiny-random-MistralForCausalLM`

If the test is fast (<5 sec) after changing the model, we can remove the `@slow` decorator. Tests with `@slow` are only run in our daily CI, and we're not as diligent in making those tests green